README.md

```diff
@@ -85,6 +88,60 @@ public class Startup
 }
 ```
 
+### Customise MongoDB persistence options and _id generation
+
+In addition to `MongoDbIdentifiable` your resource classes are free to use any of the MongoDB driver persistence options or inherit from their own base class.
+
+For example, you could change the example above so that the `Book` resource has string IDs rather than object ids in the DB, but still have them generated server side:
+
+```cs
+public class Book : IIdentifiable<string>
+{
+    // If Id=null generate a random string ID using the MongoDB driver
+    [BsonId(IdGenerator = typeof(StringObjectIdGenerator))]
```

Comment:
I'm open to consider changing `MongoDbIdentifiable` to use this instead of `[BsonId] [BsonRepresentation(BsonType.ObjectId)]` if @mrmkr agrees this is better. In any case, The MongoDB fluent mapping API can overrule these using [AutoMap](https://mongodb.github.io/mongo-csharp-driver/2.12/reference/bson/mapping/#automap), making it merely just a default for convenience.

Reply:
The default of `BsonType.ObjectId` may be sensible as it is the most performant in MongoDB because it is the binary representation.

Reply:
Good to know, then I think we should leave that the default.